processor/servicegraphprocessor/processor.go

```diff
@@ -269,6 +272,14 @@ func (p *serviceGraphProcessor) upsertDimensions(kind string, m map[string]strin
 	}
 }
 
+func (p *serviceGraphProcessor) upsertPeerAttributes(m []string, peers map[string]string, spanAttr pcommon.Map) {
+	for _, s := range m {
+		if v, ok := findAttributeValue(s, spanAttr); ok {
+			peers[s] = v
```

Comment:
If we're only using the first find of the list, we can break here and save some cycles per edge. `findAttributeValue` is a brute force search.